swagger_spec_validator/validator12.py

```diff
@@ -131,18 +132,24 @@ def validate_data_type(obj, model_ids, allow_arrays=True, allow_voids=False,
             if items is None:
                 raise SwaggerValidationError('"items" not found')
             validate_data_type(items, model_ids, allow_arrays=False)
-        elif typ in model_ids:
+            return
+        if typ == 'File':
+            if not allow_file:
+                raise SwaggerValidationError(
+                    'Type "File" is only valid for form parameters')
```

Comment:
as per spec, even the ` consumes field MUST be "multipart/form-data"` May be a #TODO for that.